be/src/runtime/mysql_result_writer.cpp

```diff
@@ -116,40 +110,65 @@ StatusOr<TFetchDataResultPtr> MysqlResultWriter::process_chunk(vectorized::Chunk
 
     // Step 2: convert chunk to mysql row format row by row
     {
+        TRY_CATCH_ALLOC_SCOPE_START()
         _row_buffer->reserve(128);
+        size_t current_bytes = 0;
+        int current_rows = 0;
         SCOPED_TIMER(_convert_tuple_timer);
+        auto result = std::make_unique<TFetchDataResult>();
+        auto& result_rows = result->result_batch.rows;
+        result_rows.resize(num_rows);
+
         for (int i = 0; i < num_rows; ++i) {
             DCHECK_EQ(0, _row_buffer->length());
             for (auto& result_column : result_columns) {
                 result_column->put_mysql_row_buffer(_row_buffer, i);
             }
             size_t len = _row_buffer->length();
-            _row_buffer->move_content(&result_rows[i]);
+
+            if (UNLIKELY(current_bytes + len >= max_row_buffer_size)) {
+                result_rows.resize(current_rows);
+                results.emplace_back(std::move(result));
+
+                result = std::make_unique<TFetchDataResult>();
+                result_rows = result->result_batch.rows;
+                result_rows.resize(num_rows - current_rows);
```

Comment:
Maybe resize to `num_rows-i`? since the number of the rest rows is less than `num_rows - current_rows`.

Reply:
> Maybe resize to `num_rows-i`? since the number of the rest rows is less than `num_rows - current_rows`.

yes……this is a mistake, I will fix it.